### PR TITLE
[scripts] Fix meson for mingw

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Fixes passing of standard link libraries for mingw: We want `-luuid`, not `-uuid`. Regression from #26617.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

### CODE REVIEW

**Comment by Neumann-A** [comment]
The previous code was:
```
    if(VCPKG_TARGET_IS_WINDOWS)
        set(c_winlibs "${VCPKG_DETECTED_CMAKE_C_STANDARD_LIBRARIES}")
        set(cpp_winlibs "${VCPKG_DETECTED_CMAKE_CXX_STANDARD_LIBRARIES}")
        foreach(libvar IN ITEMS c_winlibs cpp_winlibs)
            string(REGEX REPLACE "( |^)(-|/)" [[;\2]] "${libvar}" "${${libvar}}")
            string(REPLACE ".lib " ".lib;" "${libvar}" "${${libvar}}")
            vcpkg_list(REMOVE_ITEM "${libvar}" "")
            vcpkg_list(JOIN "${libvar}" "', '" "${libvar}")
            string(APPEND native_config "${libvar} = ['${${libvar}}']\n")
        endforeach()
    endif()
```

so how did this work previously on mingw?

**Comment by dg0yt** [comment]
> so how did this work previously on mingw?

I had this question, too, but no immedate answer.

But here it is now:
`"( |^)(-|/)" [[;\2]]` preserves the `-` or `/` by means of `\2`.
`"^(-|/)" ""` preserves neither `-` nor `/`.

Maybe it is broken also for MSVC?

File: scripts/cmake/vcpkg_configure_meson.cmake [hunk 1]
```diff
@@ -264,8 +264,10 @@ function(z_vcpkg_meson_setup_extra_windows_variables config_type)
     ## winlibs
     separate_arguments(c_winlibs NATIVE_COMMAND "${VCPKG_DETECTED_CMAKE_C_STANDARD_LIBRARIES}")
     separate_arguments(cpp_winlibs NATIVE_COMMAND "${VCPKG_DETECTED_CMAKE_CXX_STANDARD_LIBRARIES}")
-    list(TRANSFORM c_winlibs REPLACE "^(-|/)" "")
-    list(TRANSFORM cpp_winlibs REPLACE "^(-|/)" "")
+    if(NOT VCPKG_TARGET_IS_MINGW)
+        list(TRANSFORM c_winlibs REPLACE "^(-|/)" "")
+        list(TRANSFORM cpp_winlibs REPLACE "^(-|/)" "")
+    endif()
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_configure_meson.cmake:270]:
```suggestion
```

**Comment by Neumann-A** [comment]
> I had this question, too, but no immedate answer.

ah, no this piece of code was just to make it a list and is no longer needed. I thought it removed something but it wasn't meant to remove anything. Instead it should transform `-a /b c` into  `-a;/b;c`

**Review by Neumann-A — APPROVED** [review]

**Comment by dg0yt** [comment]
gdk-pixbuf is a baseline regression. (Build is flaky.) #27371 is an attempt to mitigate that. Which was hard to test because it needs this PR for mingw...

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks again!

**Pull request merged**